packages/app-desktop/gui/MainScreen/commands/toggleEditorFullScreen.ts

```diff
@@ -0,0 +1,64 @@
+import CommandService, {
+	CommandContext,
+	CommandDeclaration,
+	CommandRuntime,
+} from '@joplin/lib/services/CommandService';
+import { _ } from '@joplin/lib/locale';
+import setLayoutItemProps from '../../ResizableLayout/utils/setLayoutItemProps';
+import layoutItemProp from '../../ResizableLayout/utils/layoutItemProp';
+import { AppState } from '../../../app.reducer';
+
+// Record the current state of the sidebar and note list
+const lastUserLayoutStatus = {
+	isNoteListVisible: true,
+	isSidebarVisible: true,
+};
+
+export const declaration: CommandDeclaration = {
+	name: 'toggleEditorFullScreen',
+	label: () => _('Toggle fullscreen'),
+	iconName: 'fas fa-expand',
+};
+
+export const runtime = (): CommandRuntime => {
+	return {
+		execute: async (context: CommandContext) => {
+			const layout = (context.state as AppState).mainLayout;
+
+			const isNoteListVisible = layoutItemProp(layout, 'noteList', 'visible');
+			const isSidebarVisible = layoutItemProp(layout, 'sideBar', 'visible');
+			let isEditorFullScreen = layoutItemProp(layout, 'editor', 'isFullScreen');
+
+			if (isEditorFullScreen === undefined) {
+				isEditorFullScreen = !isNoteListVisible && !isSidebarVisible;
+			}
+
+			let newLayout = layout;
+
+			if (isEditorFullScreen) {
+				newLayout = await CommandService.instance().execute('toggleSideBar', lastUserLayoutStatus.isSidebarVisible);
+				newLayout = await CommandService.instance().execute('toggleNoteList', lastUserLayoutStatus.isNoteListVisible);
```

Comment:
I feel all this code should be refactored properly. There's no sense getting the return value fo this command if some global state is being set elsewhere.

Or maybe don't make the command return the layout if there's not really any sense to it. Just get it from the global state